core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -59,19 +64,30 @@ oppia.factory('AssetsBackendApiService', [
           }
         }
         assetsCache[filename] = audioBlob;
-        successCallback(audioBlob);
+        successCallback(AudioFileObjectFactory.createNew(filename, audioBlob));
       }).error(function() {
         errorCallback();
       })['finally'](function() {
         _removeFromFilesCurrentlyBeingRequested(filename);
       });
     };
 
+    var _abortAllCurrentDownloads = function() {
+      _filesCurrentlyBeingRequested.forEach(function(request) {
+        request.canceler.resolve();
+      });
+      _filesCurrentlyBeingRequested = [];
+    };
+
     var _removeFromFilesCurrentlyBeingRequested = function(filename) {
       if (_isCurrentlyBeingRequested(filename)) {
-        var fileToRemoveIndex =
-          _filesCurrentlyBeingRequested.indexOf(filename);
-        _filesCurrentlyBeingRequested.splice(fileToRemoveIndex, 1);
+        for (var fileToRemoveIndex = 0; fileToRemoveIndex < 
+             _filesCurrentlyBeingRequested.length; fileToRemoveIndex++) {
+          if (_filesCurrentlyBeingRequested[i].filename === filename) {
```

Comment:
Not sure what i refers to. It's OK to rename fileToRemoveIndex to index.

Can we make sure this stuff gets tested by Karma tests, btw?

Reply:
Oops, fixed. Added tests